src/elm/expression.ts

```diff
@@ -62,9 +62,14 @@ export class Expression {
 
   async execArgs(ctx: Context) {
     if (this.args != null) {
-      return Promise.all(this.args.map(async arg => arg.execute(ctx)));
+      const retVals: any[] = [];
+      for (let index = 0; index < this.args.length; index++) {
+        const arg = this.args[index];
+        retVals.push(await arg.execute(ctx));
+      }
+      return retVals;
     } else if (this.arg != null) {
-      return this.arg.execute(ctx);
+      return await this.arg.execute(ctx);
```

Comment:
Is there a reason to `await` this before returning it (given that `execArgs` is `async` itself)?

Reply:
Probably not, just felt a bit odd resolving arguments when it was an array and not when it was singular.  I removed the await though.